src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/ProjectHotReloadSessionManager.cs

```diff
@@ -199,44 +208,102 @@ public Task ActivateSessionAsync(IVsLaunchedProcess? launchedProcess, VsDebugTar
 
         async Task ActivateSessionInternalAsync()
         {
-            if (_pendingSessionState is { Session: IProjectHotReloadSession session })
+            HotReloadSessionState? sessionState = _pendingSessionState;
```

Comment:
Use the following to replace lines 211 -> 219, so that the exchange is atomic. It doesn't replace all races, but it avoids two ActivateSessionInternalAsync handling the HotReloadSessionState.
``` C#
HotReloadSessionState? sessionState = Interlocked.Exchange(ref _pendingSessionState, null);
if (sessionState is null)
{
   return;
}
```